woocommerce/woocommerce-admin
# Onboarding: Add shipping rate base task

Fixes (part of) #2482

This PR adds in the base for the shipping task.  There are still 2 more steps needed for this task, but I'll split them up as this PR is already very lengthy (sorry about that 😓 ).

* Moves the store address fields to a reusable component.
* Moves some reusable styles for onboarding under a body class `woocommerce-onboarding`.
* Fetches shipping zones for "Rest of the world" and for the local country.
* Creates or updates existing shipping methods for said zones.

### Questions/To-do

I feel there is a lot happening in this PR and I probably should have broken it up earlier.  Here are some items that I think need to be addressed (potentially in future PRs).

* Any thoughts on the folder location/naming for setting fields?  I'm thinking this long-term may be a reusable piece for WC if the settings screens gets migrated to React.
* The completion condition check for the zones seems to work here, but long-term I think we're going to keep hitting issues by not fetching updated values to check against client-side.  If we move that way, should we show a loader for the task icon while fetching any task conditions?
* The Muriel grays look to be incorrect here.  Looks like we might need to update color studio.  Does this break any designs by doing this?  It looks like most of the grays have been darkened. /cc @jameskoster 
* When returning to the shipping task after completion, I opted to allow a full walkthrough of the steps again.  After testing both ways, I feel this way feels better, but open to feedback.

### Screenshots
<img width="711" alt="Screen Shot 2019-08-07 at 6 54 14 PM" src="https://user-images.githubusercontent.com/10561050/62617709-0805f180-b945-11e9-9a4f-a4bea4ad977c.png">

### Detailed test instructions:

1.  Make sure styles are not broken in the onboarding profiler as this migrates a number of styles.  (Looks like there might be a few unrelated styling bugs as well).
2. Delete the `woocommerce_store_address` option from `wp_options` to test the store location step.
3. Make sure store address fields and validation work as expected.
4.  Delete existing zones and methods to test the task condition for completion is working.
5. Continue to the next step and make sure that updating rates creates or edits existing flat rate shipping methods (for the store default country, it will only match against zones that have **only** a single location matching the store's country).
6. Make sure that toggling the "Rest of the world" shipping zone on/off disables/enables the method on save.
